File: Application/SkuciSe/app/src/main/java/com/example/skucise/models/AdvertDetails.kt
package com.example.skucise.models

import com.example.skucise.models.HelperModels.FurnishedType
import com.example.skucise.models.HelperModels.HeatingType
import com.example.skucise.models.HelperModels.OldNewBuilding
import java.io.Serializable

class AdvertDetails(
    var id: Long,
    var date: String,
    var township: String,
    var city: String,
    var squaredArea: Int,
    var bedroomNumber: Int,
    var terrace: Boolean,
    var parking: Boolean,
    var wifi: Boolean,
    var tv: Boolean,
    var airCondition: Boolean,
    var kitchen: Boolean,
    var bathroom: Boolean,
    var alarm: Boolean,
    var pool: Boolean,
    var barbecue: Boolean,
    var firePlace: Boolean,
    var gym: Boolean,
    var description: String,
    var houseOrder: String,
    var floorNumber: Int,

    var advertId: Long,
    var furnishedTypeId: Long,
    var furnishedType: FurnishedType?,
    var heatingTypeId: Long,
    var heatingType: HeatingType?,
    var oldNewBuildingId: Long,
    var oldNewBuilding: OldNewBuilding?,
    var latitude: Double?,
    var longitude: Double?
) : Serializable{ constructor(): this(
   0,
    "",
    "",
    "",
    0,
    0,
    false,
    false,
    false,
    false,
    false,
    false,
    false,
    false,
    false,
    false,
    false,
    false,
    "",
    "",
    0,
   0,
    0,
    null,
    0,
    null,
    0,
    null,
    44.8167,
    20.4667
)
}

File: Application/Backend/SkuciSeAPI/SkuciSeAPI/Models/AdvertDetails.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace SkuciSeAPI.Models
{
    public class AdvertDetails
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Required]
        public long Id { get; set; }

        public DateTime Date { get; set; }
        public string Township { get; set; }
        public string City { get; set; }
        public long SquaredArea { get; set; }
        public long BedroomNumber { get; set; }

        public bool Terrace { get; set; }
        public bool Parking { get; set; }
        public bool Wifi { get; set; }
        public bool Tv { get; set; }
        public bool AirCondition { get; set; }
        public bool Kitchen { get; set; }
        public bool Bathroom { get; set; }
        public bool Alarm { get; set; }
        public bool Pool { get; set; }
        public bool Barbecue { get; set; }
        public bool FirePlace { get; set; }
        public bool Gym { get; set; }

        public int FloorNumber { get; set; }
        public string Description { get; set; }
        public string HouseOrder { get; set; }

        // Foreign Keys
        [ForeignKey("Advert")]
        public long AdvertId { get; set; }
        public Advert Advert { get; set; }

        public long FurnishedTypeId { get; set; }
        public FurnishedType FurnishedType { get; set; }

        public long HeatingTypeId { get; set; }
        public HeatingType HeatingType { get; set; }

        public long OldNewBuildingId { get; set; }
        public OldNewBuilding OldNewBuilding { get; set; }
        public double Latitude { get; set; }

        public double Longitude { get; set; }

    }
}
